include/chainbase/pinnable_mapped_file.hpp

```diff
@@ -1,10 +1,16 @@
 #pragma once
 
+// for BSD we should avoid any pthread calls such as pthread_mutex_lock
+#if defined(__FreeBSD__)
+#define BOOST_INTERPROCESS_FORCE_GENERIC_EMULATION
```

Comment:
Any tradeoffs adding this here vs in CMake? Sometimes I get worried of introducing an ODR violation when a different .cpp file that doesn't include this .hpp uses BIP, and thus does not have `BOOST_INTERPROCESS_FORCE_GENERIC_EMULATION` defined. But I didn't study `BOOST_INTERPROCESS_FORCE_GENERIC_EMULATION` closely to see if that's a concern or not.